### ISSUE TITLE
Mutation variables named after JavaScript keywords produce invalid Apollo Mocks

### ISSUE THREAD

**Issue by ccblaisdell** [rubengrill/apollo-typed-documents#56]
Hey rubengrill, thank you so much for this library! It's a fabulous tool, and I think it will fit in wonderfully with my workflow.

In my company's app we have a mutation that uses an argument named `delete`, which happens to also be a JavaScript reserved word. GraphQL and Apollo are fine with this, but the codegenApolloMock plugin produces invalid JavaScript when it builds mocks for this mutation.

For example, this mutation has an argument named 'delete' which accepts a list of IDs.

```gql
type Mutation {
  deleteAuthors(delete: [ID!]): [ID!]
}
```

And here is a document that uses the mutation:

```gql
mutation deleteAuthors($delete: [ID!]) {
  deleteAuthors(delete: $delete)
}
```

The generated mocks for that mutation end up like this. The `delete`s are invalid and the entire file, with all of the other generated mocks, will not be able to execute.

```js
operations.deleteAuthors = {};
operations.deleteAuthors.variables = (values = {}, options = {}) => {
  const __typename = '';
  values = (({ delete = undefined }) => ({ delete }))(values);
  //           ^^^^^^                      ^^^^^^
  values.__typename = __typename;
  return {
    delete: values.delete
  };
};
```

I have a proposed solution, which I'll link shortly, but just wanted to bring attention to this problem.

**Comment by rubengrill** [comment]
Hi @ccblaisdell, your PR is merged and a new version 2.0.1 is available (https://github.com/rubengrill/apollo-typed-documents/releases/tag/v2.0.1).

Thanks again for your PR, great work :)

**Comment by ccblaisdell** [comment]
🙌  Thanks for being receptive, and for the nuanced feedback!
